RSConnectDIOToSnap7/CodeBehind.cs

```diff
@@ -52,21 +52,21 @@ public class CodeBehind : SmartComponentCodeBehind
 		private S7Client client;
 		private bool bCanConnect = false;
 		private bool bPLC_AddrIsValid = false;
-		private bool[] bDI_AddressIsValid = { };
-		private bool[] bDO_AddressIsValid = { };
+        private bool[][] bDIO_AddressIsValid = new bool[2][];

```

Comment:
I prefer you use IO enum size to know 2 is it:
private bool[][] bDIO_AddressIsValid = new bool[((IO[])Enum.GetValues(typeof(IO))).Distinct().Count()][];
IO need to be define before and to add
using System.Linq;
after System.Collections.Generic;

Reply:
hmm, this is not the best for readability, but i also dont know better at the moment